app/models/job_invocation.rb

```diff
@@ -42,4 +45,48 @@ def failed_hosts
   def locks_for_resource(tasks, resource_type)
     tasks.map { |task| task.locks.where(:resource_type => resource_type).first }.compact
   end
+
+  def total_hosts_count
+    if targeting.resolved_at.nil?
+      _('N/A')
+    else
+      targeting.hosts.count
+    end
+  end
+
+  def delay_options
+    {
+      :start_at => start_at_parsed,
+      :start_before => start_before_parsed
+    }
+  end
+
+  def trigger_mode
+    @trigger_mode || :immediate
+  end
+
+  def trigger_mode=(value)
+    return @trigger_mode if @trigger_mode
+    if %w(immediate future).include?(value)
+      @trigger_mode = value.to_sym
+    else
+      raise ::Foreman::Exception, _("Job Invocation trigger mode must be one of 'immediate', 'future'")
```

Comment:
Would be better to have the possible values listed in some constant and use the values both in condition and exception